symfony/plugins/orangehrmTimePlugin/Service/TimesheetService.php

```diff
@@ -563,9 +570,58 @@ public function extractStartDateAndEndDateFromDate(DateTime $date): array
      * @param DateTime $date
      * @return bool
      */
-    public function isTimesheetTakenByDate(DateTime $date)
+    public function hasTimesheetForDate(DateTime $date): bool
     {
         list($startDate) = $this->extractStartDateAndEndDateFromDate($date);
         return $this->getTimesheetDao()->hasTimesheetForStartDate(new DateTime($startDate));
     }
+
+    /**
+     * @param DateTime $startDate
+     * @param int $employeeId
+     * @return Timesheet[]
+     */
+    public function getTimesheetByStartDateAndEmployeeId(int $employeeId, DateTime $startDate): array
+    {
+        return $this->getTimesheetDao()->getTimesheetByStartDateAndEmployeeId($startDate, $employeeId);
+    }
+
+    /**
+     * @param int $loggedInEmpNumber
+     * @param DetailedTimesheet $detailedTimesheet
+     * @return array
+     */
+    public function getAllowedWorkflowsForMyTimesheet(int $loggedInEmpNumber, DetailedTimesheet $detailedTimesheet): array
+    {
+        $startDateOfTimesheet = $this->getDateTimeHelper()->formatDateTimeToYmd($detailedTimesheet->getTimesheet()->getStartDate());
+        $timesheet = $this->getTimesheetByStartDateAndEmployeeId($loggedInEmpNumber, new DateTime($startDateOfTimesheet));
+
+        $includeRoles = [];
+        if ($loggedInEmpNumber == $detailedTimesheet->getTimesheet()->getEmployee()->getEmpNumber() && $this->getUserRoleManager()->essRightsToOwnWorkflow()) {
+            $includeRoles = ['ESS'];
+        }
+
+        $allowedActions = $this->getUserRoleManager()->getAllowedActions(
+            WorkflowStateMachine::FLOW_TIME_TIMESHEET,
+            $timesheet[0]->getState(),
+            [],
+            $includeRoles,
+            [Employee::class => $detailedTimesheet->getTimesheet()->getEmployee()->getEmpNumber()]
+        );
+        return $this->getAllowedActions($allowedActions);
+    }
+
+    /**
+     * @param WorkflowStateMachine[] $workflowStateMachine
+     * @return array
+     */
+    private function getAllowedActions(array $workflowStateMachine ): array
```

Comment:
```suggestion
    private function getAllowedActions(array $allowedWorkflowActions): array
```
without using this, can't we do at the API for loop

Reply:
done